### PR TITLE
Fix IME problem - especially for CJKT.

### PR DESCRIPTION
**Pull request by iLoveBug**
Works perfect on Chrome.
Known issue for Safari: the first chinese world is duplicated.

mobiledoc-kit-IME-fix

### CODE REVIEW

**Comment by huaying** [comment]
Hi I just did some tests for your fix.
Do you know which repo is a right place to fire a fix by the way?

ezgif-1-1b8804f9c119

**Comment by iLoveBug** [comment]
> Hi I just did some tests for your fix.
> Do you know which repo is a right place to fire a fix by the way?
> 
> ezgif-1-1b8804f9c119

Not sure. 

It seems the CI check test failed. Maybe we need more test.

**Comment by iLoveBug** [comment]
> Hi I just did some tests for your fix.
> Do you know which repo is a right place to fire a fix by the way?
> 
> ezgif-1-1b8804f9c119

I found the ci test failed mainly due to the safari environment problem.
截屏2019-12-19下午5 02 08

Do you mean I should raise the pull request to the root repo bustle/mobiledoc-kit?

File: src/js/editor/event-manager.js [hunk 1]
```diff
@@ -210,6 +217,43 @@ export default class EventManager {
     this._updateModifiersFromKey(key, {isDown:false});
   }
 
+  compositionstart() {
```
Comment (cmsax) [inline src/js/editor/event-manager.js]: why not use camelCase instead?
```suggestion
  compositionStart() {
```

Comment (cmsax) [reply src/js/editor/event-manager.js]: oops, that's the event name.

**Comment by wuliaopage** [comment]
> Works perfect on Chrome.
> Known issue for Safari: the first chinese world is duplicated.
> 
> mobiledoc-kit-IME-fix


Thanks so much for PR. Can you please look into this PR and repair that checks error. Many people be appreciated for your contribution. This would be a big fix for ghost for years.

**Comment by iLoveBug** [comment]
> Thanks so much for PR. Can you please look into this PR and repair that checks error. Many people be appreciated for your contribution. This would be a big fix for ghost for years.

Frankly speaking, I have no idea about the errors on CI test. It seems it's an environment problem. Anybody can help?

截屏2020-01-29上午9 53 25
截屏2020-01-29上午9 53 56

**Comment by Dixeran** [comment]
> > Thanks so much for PR. Can you please look into this PR and repair that checks error. Many people be appreciated for your contribution. This would be a big fix for ghost for years.
> 
> Frankly speaking, I have no idea about the errors on CI test. It seems it's an environment problem. Anybody can help?
> 
> 截屏2020-01-29上午9 53 25
> 
> 截屏2020-01-29上午9 53 56

By doing some search, it seems like probably because some temporary failed of saucelab. I don't know if it is appropriate to submit it (retry the CI test) again?

**Comment by iLoveBug** [comment]
How to do the re-submit?

**Comment by bin16** [comment]
@ErisDS Hi, could you please redo the checks manually? https://travis-ci.com/github/TryGhost/mobiledoc-kit/builds/141700597#L1120 the logs says it failed to request IE11/Edge/Safari environment so that the CI test ended with 3 fails

Some lines of logs:

```
not ok 594 SL_MS_Edge - [undefined ms] - bailout
not ok 602 SL_Safari_Current - [undefined ms] - bailout
not ok 603 SL_IE_11 - [undefined ms] - bailout
The environment you requested was unavailable.
...
1..2215
# tests 2215
# pass  2204
# skip  8
# fail  3
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
The command "yarn test:ci" exited with 1.
store build cache

Done. Your build exited with 1.
```

Thanks a lot!

**Comment by kevinansfield** [comment]
Please keep non-Ghost-specific PRs on the upstream [mobiledoc-kit repository](https://github.com/bustle/mobiledoc-kit/), thank you 🙂

CI will not run on this fork. It's only used for Ghost-specific changes, otherwise it's kept in sync with the main mobiledoc-kit repo.

**Pull request closed without merging**